packages/jsii-pacmak/lib/targets/go.ts

```diff
@@ -94,3 +176,56 @@ class GoGenerator implements IGenerator {
     this.code.closeFile(file);
   }
 }
+
+/**
+ * Checks if `buildDir` includes a local go build version (with "replace"
+ * directives).
+ * @param baseDir the `dist/go` directory
+ * @returns `undefined` if not or the module directory otherwise.
+ */
+function tryFindLocalModule(baseDir: string, pkg: RootPackage) {
+  const gomodPath = path.join(baseDir, pkg.packageName, GOMOD_FILENAME);
+  if (!fs.pathExistsSync(gomodPath)) {
+    return undefined;
+  }
+
+  // read `go.mod` and check that it is for the correct module
+  const gomod = fs.readFileSync(gomodPath, 'utf-8').split('\n');
+  const isExpectedModule = gomod.find(
+    (line) => line.trim() === `module ${pkg.goModuleName}`,
+  );
+
+  if (!isExpectedModule) {
+    return undefined;
+  }
+
+  return path.resolve(path.dirname(gomodPath));
+}
+
+/**
+ * Check if we are running from inside the jsii repository, and then we want to
+ * use the local runtime instead of download from a released version.
+ */
+function tryFindLocalRuntime() {
+  try {
+    const p = require.resolve('@jsii/go-runtime/jsii-runtime-go/go.mod');
```

Comment:
The 'idiom' has been to have an actual `js` entry point for the private runtime package, and have this provide the path the relevant objects...

`require('@jsii/go-runtime').runtimePath` or something.

The "indirect" coupling with the location of specific filenames in other packages has been biting us several times in the past.

Reply:
Sure I can change that

Reply:
Done